integration-tests/test_selinux.py

```diff
@@ -0,0 +1,212 @@
+    """
+    :id: 91c9c37f-b954-4fc4-84da-a4a2a9dbee9d
+    :title: Test insights-client --register runs in unconfined_t without context change
+    :description:
+        This test verifies that when running insights-client --register as root
+        in unconfined_t SELinux context with SELinux enforcing mode, the insights-core
+        code executes in unconfined_t context without any SELinux context change
+        and without any SELinux denials.
+    :tags: Tier 1
+    :reference: https://issues.redhat.com/browse/CCT-1717
+    :steps:
+        1. Register system using subscription-manager (already done by fixture)
+        2. Save current SELinux mode
+        3. Switch system to SELinux enforcing mode
+        4. Verify current context is unconfined_t
+        5. Run insights-client --register as root
+        6. Check that insights-core code was executed in unconfined_t context
+        7. Check that there was no SELinux denial during execution
+    :expectedresults:
+        1. System is registered (handled by register_subman fixture)
+        2. Current SELinux mode is saved
+        3. SELinux is set to enforcing mode
+        4. Current context is confirmed as unconfined_t
+        5. insights-client --register completes successfully
+        6. insights-core runs in unconfined_t context (no transition to confined)
+        7. No SELinux denials are found in audit logs
+    """
+    original_mode = _get_selinux_mode()
+    if original_mode == "Disabled":
+        pytest.skip(
+            reason="Skipping test: SELinux is disabled on this system "
+            "(getenforce returned 'Disabled' or command failed)"
+        )
+
+    try:
+        # Setup: Switch to enforcing mode
+        _set_selinux_mode("Enforcing")
+        assert _get_selinux_mode() == "Enforcing"
+
+        # Verify unconfined_t context (test runs in unconfined_t by default)
+        context = _get_current_context()
+        assert "unconfined_t" in context, f"Expected unconfined_t, got: {context}"
+
+        # Test: Run insights-client in unconfined_t context and check for denials
+        start_time = time.time()
+        # Pass the unconfined_t context to ensure the command runs in that context
+        result = insights_client.run("--register", selinux_context=context)
+
+        def check_registered_with_context():
+            status_result = insights_client.run(
+                "--status", check=False, selinux_context=context
+            )
+            if status_result.returncode == 0 and any(
+                i in status_result.stdout
+                for i in ["This host is registered", "Registered"]
+            ):
+                return True
+            return False
+
+        assert loop_until(check_registered_with_context)
+        assert result.returncode == 0, f"Registration failed: {result.returncode}"
+
+        # Check no SELinux denials occurred
+        denials = _check_selinux_denials(start_time=start_time)
```

Comment:
In case of no AVCs (valid and expected state), the test is unable to check that the transition to `insights_core_t` happened and it remained in the `unconfined_t` context, so different approach needs to be chosen to test this.